### PR TITLE
Fix: .signTransaction from dapp browser web3 call should only sign not sign+send

### LINKED ISSUE #6351: `.signTransaction` from dapp browser should only sign, not sign + send
@hboon  aren't we should to use ConfirmType.sign when we receive `case .signTransaction(let unconfirmedTransaction)` for dapp browser?

### PR DESCRIPTION
**Pull request by hboon**
Fixes #6351

### CODE REVIEW

**Review by oa-s — DISMISSED** [review]
uniswap looks broken with this pr

File: AlphaWallet/Browser/Coordinators/DappBrowserCoordinator.swift [hunk 1]
```diff
@@ -499,7 +499,7 @@ extension DappBrowserCoordinator: BrowserViewControllerDelegate {
         func performDappAction(account: AlphaWallet.Address) {
             switch action {
             case .signTransaction(let unconfirmedTransaction):
-                executeTransaction(action: action, callbackID: callbackID, transaction: unconfirmedTransaction, type: .signThenSend)
+                executeTransaction(action: action, callbackID: callbackID, transaction: unconfirmedTransaction, type: .sign)
```
[the remark this answers is not in the document]
Comment (oa-s) [reply AlphaWallet/Browser/Coordinators/DappBrowserCoordinator.swift:502]: we might need to add `webViewConfig.userContentController.add(messageHandler, name: Method.sendTransaction.rawValue)` as there is no such line in addition there might be need to replace processTransaction to make it send `sendTransaction` insteead of signTransaction:
```
                   processTransaction: function (tx, cb){
                       console.log('sending a transaction', tx)
                       const { id = 8888 } = tx
                       AlphaWallet.addCallback(id, cb)
                       webkit.messageHandlers.sendTransaction.postMessage({"name": "sendTransaction", "object":     tx, id: id})
                   },

```

[the remark this answers is not in the document]
Comment (oa-s) [reply AlphaWallet/Browser/Coordinators/DappBrowserCoordinator.swift:502]: and also need to define `processSignTransaction` for AlphaWallet js provider, looks like it not visible, the is required js file changes

[the remark this answers is not in the document]
Comment (oa-s) [reply AlphaWallet/Browser/Coordinators/DappBrowserCoordinator.swift:502]: @hboon do we have example dapp for testing, looks like uniswap don't work with goerli anymore, (at list for me)

[the remark this answers is not in the document]
Comment (hboon) [reply AlphaWallet/Browser/Coordinators/DappBrowserCoordinator.swift:502]: Not that I know of now. I can dig into this issue/PR if you like. Might build a test dapp.

**Comment by hboon** [comment]
Can you help reject the PR first? Just in case. Clearer :)

**Comment by oa-s** [comment]
> Can you help reject the PR first? Just in case. Clearer :)

am i did right?

**Comment by hboon** [comment]
Seems ok. GitHub seems a little strange. Not what I expected. But looks alright. Thanks.

**Comment by hboon** [comment]
Apparently we don't support `eth_signTransaction` (which is sign, but not send) in the dapp browser. We only support `eth_sendTransaction` (which is sign+send). Created https://github.com/AlphaWallet/AlphaWallet-web3-provider/issues/36. Will delay it unless it's necessary for Uniswap.

Can you help to verify that it Uniswap works with master (if it does, can you list the steps to reproduce?). If it does and this PR breaks it, then we'll fix it, otherwise, I'll clean this PR up and try to merge it again

File: AlphaWallet/Browser/Coordinators/DappBrowserCoordinator.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (hboon) [reply AlphaWallet/Browser/Coordinators/DappBrowserCoordinator.swift:502]: When testing against Uniswap, I use Polygon — cheaper :P

**Comment by hboon** [comment]
Oh I checked the PR, and looks like if we rebase master, it already includes the changes in this PR, i.e. `.signThenSend` -> `.sign` for `.signTransaction` since the handling for signing and sending transaction are now centralized with the recent refactor.

So we can close this PR once we figure out if Uniswap is broken for us.

**Comment by oa-s** [comment]
@hboon looks like this can be closed, isn't it?

**Comment by hboon** [comment]
Yes. Seems so. I’ll take a look to be sure.

**Pull request closed without merging**